graylog2-web-interface/src/components/bootstrap/BootstrapModalForm.jsx

```diff
@@ -18,12 +18,12 @@ class BootstrapModalForm extends Component {
     this.open = this.open.bind(this);
     this.close = this.close.bind(this);
     this._submit = this._submit.bind(this);
-    this._onModalClose = this._onModalClose.bind(this);
+    this._onModalCancel = this._onModalCancel.bind(this);
   }
 
-  _onModalClose() {
-    if (typeof this.props.onModalClose === 'function') {
-      this.props.onModalClose();
+  _onModalCancel() {
+    if (typeof this.props.onCancel === 'function') {
```

Comment:
There's one example that came into my mind as I read your comment: editing emails. Few apps nowadays will delete the text you wrote even if you close the window or the application. In most situations in our app, the user just need to fill in a name and description, and it's not a big issue, but in others it may need to add passwords, or even big chunks of text.

This change is specially tricky, because we are doing different things when closing or cancelling the submit, which is in my opinion a technicality, and will lead to error and frustration.

I'll also add a quote from Jef Raskin, author of The Humane Interface:

> The system should treat all user input as sacred.

Regarding the the `onModalCancel`, I missed that, so ✅